packages/amplify_authenticator/example/integration_test/sign_up_with_username_test.dart

```diff
@@ -45,106 +48,69 @@ void main() {
-      expect(find.byKey(keyEmailSignUpFormField), findsOneWidget);
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectEmailIsPresent();
     });
 
     // Scenario: "Phone Number" is not included
     testWidgets('"Phone Number" is not included', (tester) async {
-      await tester.loadAuthenticator(authenticator);
-      await tester.navigateToSignUp();
-
-      // Then I don't see "Phone Number"
-      expect(find.byKey(keyPhoneNumberSignUpFormField), findsNothing);
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectPhoneIsNotPresent();
     });
 
     // Scenario: Sign up a new username & password
     testWidgets('Sign up a new username & password', (tester) async {
-      await tester.loadAuthenticator(authenticator);
-      await tester.navigateToSignUp();
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      ConfirmSignUpPage confirmSignUpPage = ConfirmSignUpPage(tester: tester);
+
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
 
+      // TODO: Clarify requirements
       // Given I intercept '{ "headers": { "X-Amz-Target": "AWSCognitoIdentityProviderService.SignUp" } }'
       // with fixture "sign-up-with-username"
-      // TODO: Clarify requirements
 
-      // When I type a new "username"
-      await tester.enterText(
-        find.byKey(keyUsernameSignUpFormField),
-        generateUsername(),
-      );
-
-      // And I type my password
+      final username = generateUsername();
       final password = generatePassword();
-      await tester.enterText(
-        find.byKey(keyPasswordSignUpFormField),
-        password,
-      );
+      final email = generateEmail();
 
-      // And I confirm my password
-      await tester.enterText(
-        find.byKey(keyPasswordConfirmationSignUpFormField),
-        password,
-      );
-
-      // And I type my "email" with status "UNCONFIRMED"
-      await tester.enterText(
-        find.byKey(keyEmailSignUpFormField),
-        generateEmail(),
-      );
-
-      // And I type a new "preferred username"
-      await tester.enterText(
-        find.byKey(keyPreferredUsernameSignUpFormField),
-        generateUsername(),
-      );
+      await signUpPage.enterUsername(username);
+      await signUpPage.enterPassword(password);
+      await signUpPage.enterPasswordConfirmation(password);
+      await signUpPage.enterEmail(email);
+      await signUpPage.enterPreferredUsername(username);
```

Comment:
Not sure it matters, but I would probably make this a separate username than above